src/scripts/directives.js

```diff
@@ -325,19 +327,30 @@
                         scope.hourChoices = _.range(scope.hourMin, scope.hourMax, scope.hourStep);
                         scope.minuteChoices = _.range(scope.minuteMin, scope.minuteMax, scope.minuteStep);
                         var group = ['hours', 'minutes'];
+
+                        var cast = function (value) {
+                            if (scope.type === 'duration') {
+                                return moment.duration(value);
+                            }
+
+                            return moment(value);
+                        };
+
                         scope.$watch('[hours, minutes]', function (newValues) {
                             if (_.all(newValues, angular.isDefined)) {
                                 var values = _.object(_.map(group, function (key) {
                                     return [key, scope[key] || 0];
                                 }));
-                                scope.model = moment.duration(values);
+
+                                scope.model = cast(values);
+
                             }
                         }, true);
                         scope.$watch('model', function (value) {
+                            value = cast(value);
```

Comment:
Shouldn't this be inside the `if (angular.isDefined(value)) {`? I would expect `cast(undefined)` may fail or return something that is `defined`.